src/pubsub/ua_pubsub_reader.c

```diff
@@ -1953,14 +1979,84 @@ decodeAndProcessNetworkMessage(UA_Server *server, UA_ReaderGroup *readerGroup,
                  (size_t)(connection->config->transportProfileUri.length)) == 0)) {
         paddingBytes = (MIN_PAYLOAD_SIZE_ETHERNET - ((*currentPosition) - previousPosition));
         (*currentPosition) += paddingBytes; /* During multiple receive, move the position to
-                             handle padding bytes */
+                                               handle padding bytes */
     }
 
 cleanup:
     UA_NetworkMessage_clear(&currentNetworkMessage);
     return rv;
 }
 
+static UA_StatusCode
+decodeHeadersAndDecryptNM(UA_Server *server, UA_ReaderGroup *readerGroup, UA_ByteString *buffer) {
+
+    UA_NetworkMessage currentNetworkMessage;
+    memset(&currentNetworkMessage, 0, sizeof(UA_NetworkMessage));
+    size_t payLoadPosition = 0;
+    UA_StatusCode rv = UA_NetworkMessage_decodeHeaders(buffer, &payLoadPosition, &currentNetworkMessage);
+
+    UA_CHECK_STATUS_ERROR(rv, return rv, &server->config.logger, UA_LOGCATEGORY_SERVER,
+                          "PubSub receive. decoding headers failed");
+
+#ifdef UA_ENABLE_PUBSUB_ENCRYPTION
+    if(readerGroup->config.rtLevel == UA_PUBSUB_RT_FIXED_SIZE) {
+        if(readerGroup->config.securityMode > UA_MESSAGESECURITYMODE_NONE) {
+            rv = verifyAndDecryptNetworkMessage(&server->config.logger,
+                                                buffer,
+                                                &payLoadPosition,
+                                                &currentNetworkMessage,
+                                                readerGroup);
+            UA_CHECK_STATUS_WARN(rv, return rv, &server->config.logger, UA_LOGCATEGORY_SERVER,
+                                 "Subscribe failed. verify and decrypt network message failed.");
+            UA_NetworkMessage_clear(&currentNetworkMessage);
+            return rv;
+        }
+        else{
+            return UA_STATUSCODE_BADREQUESTTYPEINVALID;
+        }
```

Comment:
End the if-case with return UA_STATUSCODE_GOOD.
That saves one level of indendation instead of having an else clause.